### PR TITLE
Move gregified recipes from GC to coremod

### PR DESCRIPTION
**Pull request by HoleFish**
Hopefully GT hard dep is removed
https://github.com/GTNewHorizons/NewHorizonsCoreMod/pull/889

### CODE REVIEW

File: src/main/java/micdoodle8/mods/galacticraft/core/nei/NEIGalacticraftConfig.java [hunk 1]
```diff
@@ -140,121 +134,56 @@ public void registerRecipes() {
...
+         * ItemStack(GCItems.heavyPlatingTier1), 80 - x, 91 - y)); this.registerBuggyBenchRecipe(input, new
+         * PositionedStack(new ItemStack(GCItems.buggy), 143 - x, 64 - y)); HashMap<Integer, PositionedStack> input2 =
+         * new HashMap<>(input); input2.put(35, new PositionedStack(RecipeUtil.getChestItemStack(1, 3), 107 - x, 64 -
+         * y)); this.registerBuggyBenchRecipe(input2, new PositionedStack(new ItemStack(GCItems.buggy, 1, 1), 143 - x,
+         * 64 - y)); input2 = new HashMap<>(input); input2.put(35, new PositionedStack(RecipeUtil.getChestItemStack(1,
+         * 0), 107 - x, 64 - y)); this.registerBuggyBenchRecipe(input2, new PositionedStack(new ItemStack(GCItems.buggy,
+         * 1, 2), 143 - x, 64 - y)); input2 = new HashMap<>(input); input2.put(35, new
+         * PositionedStack(RecipeUtil.getChestItemStack(1, 1), 107 - x, 64 - y)); this.registerBuggyBenchRecipe(input2,
+         * new PositionedStack(new ItemStack(GCItems.buggy, 1, 3), 143 - x, 64 - y));
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/core/nei/NEIGalacticraftConfig.java]: remove the dead code, git history is here if it is needed again.

File: src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java [hunk 2]
```diff
@@ -209,44 +199,22 @@ private static void addUniversalRecipes() {
...
+         * <= 16; i++) { input.put(i, new ItemStack(GCItems.meteoricIronIngot, 1, 1)); } for (int i = 17; i <= 24; i++)
+         * { input.put(i, GT_OreDictUnificator.get(OrePrefixes.screw, Materials.StainlessSteel, 1)); } for (int i = 25;
+         * i <= 34; i++) { input.put(i, new ItemStack(GCItems.heavyPlatingTier1)); } input2 = new HashMap<>(input);
+         * input2.put(35, null); RecipeUtil.addBuggyBenchRecipe(new ItemStack(GCItems.buggy, 1, 0), input2); input2 =
+         * new HashMap<>(input); input2.put(35, RecipeUtil.getChestItemStack(1, 3)); RecipeUtil.addBuggyBenchRecipe(new
+         * ItemStack(GCItems.buggy, 1, 1), input2); input2 = new HashMap<>(input); input2.put(35,
+         * RecipeUtil.getChestItemStack(1, 0)); RecipeUtil.addBuggyBenchRecipe(new ItemStack(GCItems.buggy, 1, 2),
+         * input2); input2 = new HashMap<>(input); input2.put(35, RecipeUtil.getChestItemStack(1, 1));
+         * RecipeUtil.addBuggyBenchRecipe(new ItemStack(GCItems.buggy, 1, 3), input2);
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java [hunk 3]
```diff
@@ -278,14 +246,13 @@ private static void addUniversalRecipes() {
...
-        GalacticraftRegistry.registerSpaceStation(
-                new SpaceStationType(ConfigManagerCore.idDimensionOverworldOrbit, 0, new SpaceStationRecipe(inputMap)));
-
+        /*
+         * final HashMap<Object, Integer> inputMap = new HashMap<>(); inputMap.put(new ItemStack(GCBlocks.basicBlock, 1,
+         * 4), 231); inputMap.put(new ItemStack(Blocks.glass_pane), 6); inputMap.put("circuitAdvanced", 4);
+         * inputMap.put(new ItemStack(GregTech_API.sBlockMachines, 1, 13), 1); // HV Machine Hull
+         * GalacticraftRegistry.registerSpaceStation( new SpaceStationType(ConfigManagerCore.idDimensionOverworldOrbit,
+         * 0, new SpaceStationRecipe(inputMap)));
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/AsteroidsModule.java [hunk 4]
```diff
@@ -196,7 +190,7 @@ public void init(FMLInitializationEvent event) {
         GalacticraftRegistry.registerTeleportType(WorldProviderAsteroids.class, new TeleportTypeAsteroids());
 
         // Handled by Galaxy Space
-        final HashMap<Integer, ItemStack> input = new HashMap<>();
+        // final HashMap<Integer, ItemStack> input = new HashMap<>();
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/AsteroidsModule.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/AsteroidsModule.java [hunk 5]
```diff
@@ -233,34 +227,21 @@ public void init(FMLInitializationEvent event) {
...
+         * GT_ModHandler.getModItem(Constants.MOD_ID_GALAXYSPACE, "item.RocketControlComputer", 1, 102)); }
+         * input.put(19, new ItemStack(GCItems.basicItem, 1, 14)); input.put(20, new ItemStack(GCItems.basicItem, 1,
+         * 14)); for (int i = 21; i <= 23; i++) { input.put(i, new ItemStack(GCItems.heavyPlatingTier1)); }
+         * input.put(24, RecipeUtil.getChestItemStack(1, 1)); input.put(25, RecipeUtil.getChestItemStack(1, 1));
+         * input.put(26, new ItemStack(AsteroidsItems.basicItem, 1, 8)); input.put(27, new
+         * ItemStack(AsteroidBlocks.beamReceiver)); input.put(28, GT_ModHandler.getModItem(Constants.MOD_ID_GREGTECH,
+         * "gt.metaitem.01", 1, 32603)); input.put(29, GT_ModHandler.getModItem(Constants.MOD_ID_GREGTECH,
+         * "gt.metaitem.01", 1, 32603)); GalacticraftRegistry .addAstroMinerRecipe(new NasaWorkbenchRecipe(new
+         * ItemStack(AsteroidsItems.astroMiner, 1, 0), input));
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/AsteroidsModule.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/nei/NEIGalacticraftAsteroidsConfig.java [hunk 6]
```diff
@@ -63,7 +55,7 @@ public static Set<Map.Entry<ArrayList<PositionedStack>, PositionedStack>> getAst
     }
 
     public void registerRecipes() {
-        ArrayList<PositionedStack> input = new ArrayList<>();
+        // ArrayList<PositionedStack> input = new ArrayList<>();
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/nei/NEIGalacticraftAsteroidsConfig.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/nei/NEIGalacticraftAsteroidsConfig.java [hunk 7]
```diff
@@ -110,57 +102,34 @@ public void registerRecipes() {
...
+         * ItemStack(GCItems.heavyPlatingTier1), 116 - x, 73 - y)); input.add(new
+         * PositionedStack(RecipeUtil.getChestItemStack(1, 1), 80 - x, 55 - y)); input.add(new
+         * PositionedStack(RecipeUtil.getChestItemStack(1, 1), 98 - x, 55 - y)); input.add(new PositionedStack(new
+         * ItemStack(AsteroidsItems.basicItem, 1, 8), 44 - x, 73 - y)); input.add(new PositionedStack(new
+         * ItemStack(AsteroidBlocks.beamReceiver), 62 - x, 73 - y)); input.add( new PositionedStack(
+         * GT_ModHandler.getModItem(Constants.MOD_ID_GREGTECH, "gt.metaitem.01", 1, 32603), 80 - x, 73 - y)); input.add(
+         * new PositionedStack( GT_ModHandler.getModItem(Constants.MOD_ID_GREGTECH, "gt.metaitem.01", 1, 32603), 98 - x,
+         * 73 - y)); this.registerAstroMinerRecipe( input, new PositionedStack(new ItemStack(AsteroidsItems.astroMiner),
+         * 143 - x, 55 - y));
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/nei/NEIGalacticraftAsteroidsConfig.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/mars/nei/NEIGalacticraftMarsConfig.java [hunk 8]
```diff
@@ -92,7 +89,7 @@ public void registerRecipes() {
         // Handled by GalaxySpace
         // final int changeY = 15;
 
-        final ArrayList<PositionedStack> input = new ArrayList<>();
+        // final ArrayList<PositionedStack> input = new ArrayList<>();
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/mars/nei/NEIGalacticraftMarsConfig.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/mars/nei/NEIGalacticraftMarsConfig.java [hunk 9]
```diff
@@ -116,112 +113,56 @@ public void registerRecipes() {
...
+         * 80 - x, 91 - y)); input.add(new PositionedStack(new ItemStack(GCItems.partFins), 26 - x, 109 - y));
+         * input.add(new PositionedStack(new ItemStack(GCItems.partFins), 80 - x, 109 - y)); input2 = new
+         * ArrayList<>(input); input2.add(new PositionedStack(RecipeUtil.getChestItemStack(1, 3), 134 - x, 46 - y));
+         * this.registerCargoBenchRecipe( input2, new PositionedStack(new ItemStack(MarsItems.spaceship, 1, 11), 134 -
+         * x, 73 - y)); input2 = new ArrayList<>(input); input2.add(new PositionedStack(RecipeUtil.getChestItemStack(1,
+         * 0), 134 - x, 46 - y)); this.registerCargoBenchRecipe( input2, new PositionedStack(new
+         * ItemStack(MarsItems.spaceship, 1, 12), 134 - x, 73 - y)); input2 = new ArrayList<>(input); input2.add(new
+         * PositionedStack(RecipeUtil.getChestItemStack(1, 1), 134 - x, 46 - y)); this.registerCargoBenchRecipe( input2,
+         * new PositionedStack(new ItemStack(MarsItems.spaceship, 1, 13), 134 - x, 73 - y));
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/mars/nei/NEIGalacticraftMarsConfig.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/mars/recipe/RecipeManagerMars.java [hunk 10]
```diff
@@ -211,7 +205,7 @@ private static void addUniversalRecipes() {
 
         // Handled by Galaxy Space
         // Schematic
-        final HashMap<Integer, ItemStack> input = new HashMap<>();
+        // final HashMap<Integer, ItemStack> input = new HashMap<>();
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/mars/recipe/RecipeManagerMars.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/mars/recipe/RecipeManagerMars.java [hunk 11]
```diff
@@ -227,56 +221,41 @@ private static void addUniversalRecipes() {
...
+         * input2.put(20, new ItemStack(Blocks.chest)); input2.put(21, null); MarsUtil.addRocketBenchT2Recipe(new
+         * ItemStack(MarsItems.spaceship, 1, 2), input2); input2 = new HashMap<Integer, ItemStack>(input);
+         * input2.put(19, new ItemStack(Blocks.chest)); input2.put(20, null); input2.put(21, new
+         * ItemStack(Blocks.chest)); MarsUtil.addRocketBenchT2Recipe(new ItemStack(MarsItems.spaceship, 1, 2), input2);
+         * input2 = new HashMap<Integer, ItemStack>(input); input2.put(19, null); input2.put(20, new
+         * ItemStack(Blocks.chest)); input2.put(21, new ItemStack(Blocks.chest)); MarsUtil.addRocketBenchT2Recipe(new
+         * ItemStack(MarsItems.spaceship, 1, 2), input2); input2 = new HashMap<Integer, ItemStack>(input);
+         * input2.put(19, new ItemStack(Blocks.chest)); input2.put(20, new ItemStack(Blocks.chest)); input2.put(21, new
+         * ItemStack(Blocks.chest)); MarsUtil.addRocketBenchT2Recipe(new ItemStack(MarsItems.spaceship, 1, 3), input2);
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/mars/recipe/RecipeManagerMars.java]: same here

File: src/main/java/micdoodle8/mods/galacticraft/planets/mars/recipe/RecipeManagerMars.java [hunk 12]
```diff
@@ -227,56 +221,41 @@ private static void addUniversalRecipes() {
...
+         * "item.ModuleSmallFuelCanister", 1)); } } input.put(7, new ItemStack(GCItems.partNoseCone)); for (int i = 8; i
+         * <= 15; i++) { input.put(i, new ItemStack(MarsItems.marsItemBasic, 1, 3)); } input.put(16, new
+         * ItemStack(GCItems.rocketEngine)); for (int i = 17; i <= 20; i++) { input.put(i, new
+         * ItemStack(GCItems.partFins)); } input2 = new HashMap<>(input); input2.put(21, RecipeUtil.getChestItemStack(1,
+         * 3)); MarsUtil.adCargoRocketRecipe(new ItemStack(MarsItems.spaceship, 1, 11), input2); input2 = new
+         * HashMap<>(input); input2.put(21, RecipeUtil.getChestItemStack(1, 0)); MarsUtil.adCargoRocketRecipe(new
+         * ItemStack(MarsItems.spaceship, 1, 12), input2); input2 = new HashMap<>(input); input2.put(21,
+         * RecipeUtil.getChestItemStack(1, 1)); MarsUtil.adCargoRocketRecipe(new ItemStack(MarsItems.spaceship, 1, 13),
+         * input2);
+         */
```
Comment (boubou19) [inline src/main/java/micdoodle8/mods/galacticraft/planets/mars/recipe/RecipeManagerMars.java]: same here

**Review by boubou19 — DISMISSED** [review]

**Comment by boubou19** [comment]
Should be undrafted only when the corresponding PR to NHCore will be made and linked here

File: src/main/java/micdoodle8/mods/galacticraft/core/nei/NEIGalacticraftConfig.java (hunk 1 above)
Comment (HoleFish) [reply src/main/java/micdoodle8/mods/galacticraft/core/nei/NEIGalacticraftConfig.java]: Shall we remove other dead code?

Comment (boubou19) [reply src/main/java/micdoodle8/mods/galacticraft/core/nei/NEIGalacticraftConfig.java]: yes, unless there is explicitely a comment saying why it should stay

File: src/main/java/micdoodle8/mods/galacticraft/core/inventory/SlotBuggyBench.java [hunk 13]
```diff
@@ -68,12 +69,12 @@ public boolean isItemValid(ItemStack itemStack) {
                     && itemStack.getItemDamage() == 100;
         } else if (this.index >= 4 && this.index <= 7) {
             return itemStack.getItem() == GCItems.partBuggy && itemStack.getItemDamage() == 0;
-        } else if (this.index >= 8 && this.index <= 11) {
+        } else if (this.index >= 8 && this.index <= 11 && Loader.isModLoaded(Constants.MOD_ID_GREGTECH)) {
```
Comment (chochem) [inline src/main/java/micdoodle8/mods/galacticraft/core/inventory/SlotBuggyBench.java]: just checking for gregtech also kicks in for GT6.

If you want to check for GTNH better check for dreamcraft, if you specifically want to check for GT5U-GTNH, check for gregtech but not gregapi. see for example https://github.com/GTNewHorizons/ThaumicBases/pull/36

**Review by Dream-Master — APPROVED** [review]
checked ingame. looks ok

**Pull request merged**